test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/AnchorTagHelperTest.cs

```diff
@@ -187,6 +187,178 @@ public class AnchorTagHelperTest
+                .Verifiable();
+            var anchorTagHelper = new AnchorTagHelper(generator.Object)
+            {
+                Action = "Index",
+                Controller = "Home",
+                Area = "Admin",
+                Fragment = "hello=world",
+                Host = "contoso.com",
+                Protocol = "http",
+            };
+
+            // Act
+            await anchorTagHelper.ProcessAsync(context, output);
+
+            // Assert
+            generator.Verify();
+
+            Assert.Equal("a", output.TagName);
+            Assert.Empty(output.Attributes);
+            Assert.True(output.Content.IsEmpty);
+        }
+
+        [Fact]
+        public async Task ProcessAsync_AspAreaOverridesAspRouteArea()
+        {
+            // Arrange
+            var context = new TagHelperContext(
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+            var output = new TagHelperOutput(
+                "a",
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) =>
+                {
+                    var tagHelperContent = new DefaultTagHelperContent();
+                    tagHelperContent.SetContent("Something");
+                    return Task.FromResult<TagHelperContent>(tagHelperContent);
+                });
+            output.Content.SetContent(string.Empty);
+
+            var generator = new Mock<IHtmlGenerator>();
+            var routeValues = new Dictionary<string, object> { { "area", "Admin" } };
+
+            generator
+                .Setup(mock => mock.GenerateActionLink(
+                    It.IsAny<ViewContext>(),
+                    string.Empty,
+                    "Index",
+                    "Home",
+                    "http",
+                    "contoso.com",
+                    "hello=world",
+                    routeValues,
+                    null))
+                .Returns(new TagBuilder("a"))
+                .Verifiable();
+            var anchorTagHelper = new AnchorTagHelper(generator.Object)
+            {
+                Action = "Index",
+                Controller = "Home",
+                Area = "Admin",
+                Fragment = "hello=world",
+                Host = "contoso.com",
+                Protocol = "http",
+                RouteValues = new Dictionary<string, string> { { "area", "Home" } }
+            };
+
+            // Act
+            await anchorTagHelper.ProcessAsync(context, output);
+
+            // Assert
+            generator.Verify();
+
+            Assert.Equal("a", output.TagName);
+            Assert.Empty(output.Attributes);
+            Assert.True(output.Content.IsEmpty);
+        }
+
+        [Fact]
+        public async Task ProcessAsync_EmptyStringOnAspAreaClearsContextArea()
```

Comment:
Rename this test because there's nothing getting `Clear`ed in this test. All you're demonstrating is that `string.Empty` is passed through to the `IHtmlGenerator` correctly.